modules/pam_limits/pam_limits.c

```diff
@@ -816,30 +820,18 @@ parse_config_file(pam_handle_t *pamh, const char *uname, uid_t uid, gid_t gid,
     FILE *fil;
     char buf[LINE_LENGTH];
 
-    /* check for the CONF_FILE */
+    /* check for the file */
     if (ctrl & PAM_DEBUG_ARG)
-        pam_syslog(pamh, LOG_DEBUG, "reading settings from '%s'", CONF_FILE);
-    fil = fopen(CONF_FILE, "r");
+        pam_syslog(pamh, LOG_DEBUG, "reading settings from '%s'", pl->conf_file);
+    fil = fopen(pl->conf_file, "r");
     if (fil == NULL) {
-      int err = errno;
-
-#ifdef VENDOR_SCONFIGDIR
-      /* if the specified file does not exist, and it is not provided by
-         the user, try the vendor file as fallback. */
-      if (pl->conf_file == NULL && err == ENOENT)
-        fil = fopen(VENDOR_SCONFIGDIR "/limits.conf", "r");
-
-      if (fil == NULL)
-#endif
-        {
-          if (err == ENOENT)
+        if (errno == ENOENT)
             return PAM_SUCCESS;
```

Comment:
This would treat ENOENT as PAM_SUCCESS not just for default limits.conf, but also for config files specified using `conf=` options.  This is a change in behaviour.